<a href="https://colab.research.google.com/github/Pedro-Henrique8899-eng/covid-19-data/blob/master/analisando_casos_de_covid.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Obtenção de Dados

A coleta de dados para análises tem sido um desafio e tanto durante o período pandêmico, uma vez que, as fontes para obtenção de dados são as mais variadas. Temos os dados disponiblizados pelo governo federal, imprensa, pelos estados e os dados não oficiais.

Analisar uma doença nova já é um desafio gigantesco, em um país como o Brasil onde não se tem uma cultura de dados esse desafio se torna ainda maior.

## Dicionário de Variáveis

* ...iso_code - código                                         
* ...continent - continente                                        
* ...location - localização                                        
* ...date - data                                   
* ...total_cases - total de casos                                    
* ...new_cases - novos casos                                     
* ...new_cases_smoothed - novos casos suavizados                           
* ...total_deaths - total de mortes                                   
* ...new_deaths - novas mortes                                     
* ...new_deaths_smoothed - novas mortes suavizadas                           
* ...total_cases_per_million - total de casos por milhão                        
* ...new_cases_per_million - novos casos por milhão                          
* ...new_cases_smoothed_per_million - novos casos suavizados por milhão               
* ...total_deaths_per_million - total de casos por milhão                        
* ...new_deaths_per_million - novas mortes por milhão                         
* ...new_deaths_smoothed_per_million - novas mortes suavizadas por milhão                
* ...reproduction_rate - taxa de reprodução                             
* ...icu_patients - pacientes na UTI                                   
* ...icu_patients_per_million - UTI para pacientes por milhão                    
* ...hosp_patients - pacientes de hospital                                   
* ...hosp_patients_per_million - pacientes de hospital por milhão                    
* ...weekly_icu_admissions - admissão semanal na UTI                           
* ...weekly_icu_admissions_per_million - admissão semanal na UTI por milhão              
* ...weekly_hosp_admissions - internações semanais                          
* ...weekly_hosp_admissions_per_million - internações semanais por milhão            
* ...new_tests - novos testes                                      
* ...total_tests - total de testes                                 
* ...total_tests_per_thousand - total de testes por mil                   
* ...new_tests_per_thousand - novos testes por mil                         
* ...new_tests_smoothed - novos testes suavizados                             
* ...new_tests_smoothed_per_thousand - novos testes suavizados por mil                
* ...positive_rate - taxa positiva                                  
* ...tests_per_case - testes por caso                                 
* ...tests_units - unidades de teste                                   
* ...total_vaccinations - total de vacinações                             
* ...people_vaccinated - pessoas vacinadas                              
* ...people_fully_vaccinated - pessoas totalmente vacinadas                        
* ...new_vaccinations - novas vacinações                              
* ...new_vaccinations_smoothed - novas vacinações suavizadas                      
* ...total_vaccinations_per_hundred - vacinação total por cem habitantes                 
* ...people_vaccinated_per_hundred - pessoas vacinadas por cem habitantes                  
* ...people_fully_vaccinated_per_hundred - pessoas totalmente vacinadas por cem habitantes     
* ...new_vaccinations_smoothed_per_million - novas vacinações suavizadas por milhão          
* ...stringency_index - índice de rigor                              
* ...population - população                                     
* ...population_density - densidade populacional                            
* ...median_age - idade média                                    
* ...aged_65_older - com 65 anos de idade                                
* ...aged_70_older - com 70 anos de idade                                  
* ...gdp_per_capita - PIB per capita                                
* ...extreme_poverty - pobreza extrema                 
* ...cardiovasc_death_rate - taxa de mortalidade cardiovascular                             
* ...diabetes_prevalence - prevalência de diabetes                         
* ...female_smokers - fumantes femininas                                
* ...male_smokers - fumantes masculinos                                  
* ...handwashing_facilities - instalações para lavagem das mãos                        
* ...hospital_beds_per_thousand - leitos de hospitais por  mil habitantes                       
* ...life_expectancy - expectativa de vida                               
* ...human_development_index - índice de desenvolvimento humano                         


## Importando os dados do Covid-19

*  Os dados utilizados nesta análise foram obtidos em (https://raw.githubusercontent.com/Pedro-Henrique8899-eng/data_science_sigmoidal/main/owid-covid-data.csv). Utilizaremos o formato `csv` desse arquivo.



*  A variável `df` irá receber o endereço do arquivo

   df = pd.read_csv('Endereço_do_arquivo.csv')

In [2]:
# importando as bibliotecas necessárias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
# importando o Dataset da Covid-19 para um DataFrame
df = pd.read_csv('https://raw.githubusercontent.com/Pedro-Henrique8899-eng/data_science_sigmoidal/main/owid-covid-data.csv')


## Análise Exploratória de Dados

A etapa de análise exploratória dos dados é uma etapa extremamente importante, pois será através dela que vamos obter informações referentes aos dados que estamos trabalhando e como eles estão estruturados.


## Análise Exploratória inicial

De início, faremos uma breve exploração dos dados afim de verificar algumas informações relevantes para que possamos prosseguir para uma análise um pouco mais detalhada.

In [4]:
# verificando as primeiras 5 entradas do DataFrame
print('Quantidade de linhas: {}' .format(df.shape[0]))
print('Quantidade de colunas: {}'.format(df.shape[1]))

Quantidade de linhas: 75558
Quantidade de colunas: 59


Acima podemos observar que o nosso dataset possui 75558 linhas e 59 colunas.

In [5]:
# verificando as primeiras 5 entradas
df.head() 

iso_code continent  ... life_expectancy human_development_index
0      AFG      Asia  ...           64.83                   0.511
1      AFG      Asia  ...           64.83                   0.511
2      AFG      Asia  ...           64.83                   0.511
3      AFG      Asia  ...           64.83                   0.511
4      AFG      Asia  ...           64.83                   0.511

[5 rows x 59 columns]

In [25]:
# verificando o nome das colunas
df.columns

Index(['iso_code', 'continent', 'location', 'date', 'total_cases', 'new_cases',
       'new_cases_smoothed', 'total_deaths', 'new_deaths',
       'new_deaths_smoothed', 'total_cases_per_million',
       'new_cases_per_million', 'new_cases_smoothed_per_million',
       'total_deaths_per_million', 'new_deaths_per_million',
       'new_deaths_smoothed_per_million', 'reproduction_rate', 'icu_patients',
       'icu_patients_per_million', 'hosp_patients',
       'hosp_patients_per_million', 'weekly_icu_admissions',
       'weekly_icu_admissions_per_million', 'weekly_hosp_admissions',
       'weekly_hosp_admissions_per_million', 'new_tests', 'total_tests',
       'total_tests_per_thousand', 'new_tests_per_thousand',
       'new_tests_smoothed', 'new_tests_smoothed_per_thousand',
       'positive_rate', 'tests_per_case', 'tests_units', 'total_vaccinations',
       'people_vaccinated', 'people_fully_vaccinated', 'new_vaccinations',
       'new_vaccinations_smoothed', 'total_vaccinations_per_hun

Podemos verificar todos os nomes de cada uma das colunas do nosso DataFrame.

A variável date é dada como um objeto do tipo string. Com isso, vamos transformá-la em um objeto do tipo datetime.

In [7]:
# transformando a variável date em um objeto do tipo datetime
df['date'] = pd.to_datetime(df.date)

In [8]:
# verificando os tipos das variáveis
df.dtypes

iso_code                                         object
continent                                        object
location                                         object
date                                     datetime64[ns]
total_cases                                     float64
new_cases                                       float64
new_cases_smoothed                              float64
total_deaths                                    float64
new_deaths                                      float64
new_deaths_smoothed                             float64
total_cases_per_million                         float64
new_cases_per_million                           float64
new_cases_smoothed_per_million                  float64
total_deaths_per_million                        float64
new_deaths_per_million                          float64
new_deaths_smoothed_per_million                 float64
reproduction_rate                               float64
icu_patients                                    

É importante que verifiquemos a porcentagem de valores nulos em cada uma das colunas do nosso Quadro de Dados. Faremos isso na ordem decrescente.

In [14]:
# verificando a porcentagem de valores nulos em cada coluna
(df.isnull().sum() / df.shape[0]).sort_values(ascending=False)

weekly_icu_admissions                    0.990961
weekly_icu_admissions_per_million        0.990961
weekly_hosp_admissions                   0.983311
weekly_hosp_admissions_per_million       0.983311
people_fully_vaccinated                  0.962585
people_fully_vaccinated_per_hundred      0.962585
new_vaccinations                         0.946690
people_vaccinated                        0.943950
people_vaccinated_per_hundred            0.943950
total_vaccinations                       0.937558
total_vaccinations_per_hundred           0.937558
new_vaccinations_smoothed                0.906350
new_vaccinations_smoothed_per_million    0.906350
icu_patients                             0.897311
icu_patients_per_million                 0.897311
hosp_patients                            0.876426
hosp_patients_per_million                0.876426
total_tests                              0.550742
total_tests_per_thousand                 0.550742
new_tests                                0.547831


É possível observar que uma grande quantidade das colunas possuem quase 100% dos seus valores faltantes. Outras aproximadamente a metade e algumas uma porcentagem pequena.

Apenas as colunas *iso_code*, *location* e *date* não possuem valores ausentes.

É importante nos atentarmos para o motivo dessa grande quantidade de dados estarem faltando em nosso DataSet.


 Agora vamos verificar a data mais atual quais os países tem mais mortes e casos confirmados.

 Para isso, antes iremos verificar qual a data mais atual do nosso dataset.

In [20]:
# verificando a data mais atual
df.date.max()

Timestamp('2021-03-17 00:00:00')

In [18]:
# verificando para a data mais atual quais os países que tem mais casos confirmados e mortes
df.loc[df.date == '2021-03-17'].sort_values(by='total_cases', ascending=False)
df.loc[df.date == '2021-03-17'].sort_values(by='total_deaths', ascending=False)

iso_code      continent  ... life_expectancy human_development_index
74487  OWID_WRL            NaN  ...           72.58                   0.737
23408  OWID_EUR            NaN  ...             NaN                     NaN
50114  OWID_NAM            NaN  ...             NaN                     NaN
23828  OWID_EUN            NaN  ...             NaN                     NaN
72034       USA  North America  ...           78.86                   0.926
...         ...            ...  ...             ...                     ...
57944       WSM        Oceania  ...           73.32                   0.715
61911       SLB        Oceania  ...           73.00                   0.567
68516       TLS           Asia  ...           69.50                   0.606
72900       VUT        Oceania  ...           70.47                   0.609
73277       VAT         Europe  ...           75.12                     NaN

[202 rows x 59 columns]

In [ ]:
# plotar um gráfico de barras para os 5 países com mais mortes
#df.loc[df.date == '2021-03-17', ['location', 'total_deaths']].sort_values(by='total_deaths', ascending=False)[1:6]

fig, ax = plt.subplots(figsize=(10,6))

df['location', 'total_deaths'].plot
#ax.plot(location, total_deaths, kind='bar')
#ax.set_ylabel('total_deaths')
#ax.set_xlabel('location')

#plt.tight_layout()
